exec/java-exec/src/main/java/org/apache/drill/exec/compile/ClassTransformer.java

```diff
@@ -307,26 +288,38 @@ public Class<?> getImplementationClass(
           s = s.replace(DrillFileUtils.SEPARATOR_CHAR, '.');
           names.add(nextSet.getChild(s));
         }
-        classLoader.injectByteCode(nextGenerated.dot, result.bytes);
+
+        // Only inject bytecode if not already injected
+        if (!injectedClassNames.contains(nextGenerated.dot)) {
```

Comment:
It's hard to tell from the diff what was broken here before. Was the `namesCompleted` Set not enough to keep track of classes that had already been merged?

Reply:
I'm not sure honestly.  It seemed to break when I switched the Java version to 11 in the test suite.